media-sound/ncmpcpp/ncmpcpp-0.9.2_p20240724.ebuild

```diff
@@ -0,0 +1,68 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools
+
+NCMPCPP_COMMIT="68daf44032784a5b5b74781a9bf3826053eff4a2"
+
+DESCRIPTION="Featureful ncurses based MPD client inspired by ncmpc"
+HOMEPAGE="https://rybczak.net/ncmpcpp/"
+SRC_URI="https://github.com/ncmpcpp/ncmpcpp/archive/${NCMPCPP_COMMIT}.tar.gz -> ${P}.gh.tar.gz"
+S="${WORKDIR}/${PN}-${NCMPCPP_COMMIT}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="clock network outputs taglib visualizer"
+
+RDEPEND="
+	>=media-libs/libmpdclient-2.1
+	dev-libs/boost:=[icu,nls]
```

Comment:
Why are you removing `dev-libs/icu:=` as dependency? From what I see, ncmpcpp needs it:

```
$ objdump -p /usr/bin/ncmpcpp | grep NEED
  NEEDED               libboost_filesystem.so.1.85.0
  NEEDED               libboost_locale.so.1.85.0
  NEEDED               libboost_program_options.so.1.85.0
  NEEDED               libboost_thread.so.1.85.0
  NEEDED               libicui18n.so.74
  NEEDED               libicuuc.so.74
  NEEDED               libmpdclient.so.2
  NEEDED               libreadline.so.8
  NEEDED               libncursesw.so.6
  NEEDED               libtinfow.so.6
  NEEDED               libcurl.so.4
  NEEDED               libstdc++.so.6
  NEEDED               libgcc_s.so.1
  NEEDED               libc.so.6
  VERNEED              0x000000000000b9f0
  VERNEEDNUM           0x0000000000000004
```

Also, looking at `configure.ac`, there is a call `PKG_CHECK_MODULES([ICU], [icu-i18n icu-uc], [ (...)`

Same goes for `net-libs/curl`. `configure.ac` considers a missing libcurl a fatal error, so this should probably be an unconditional dependency.

Reply:
This is called a transitive dependency. While in this particular case you could be quite sure that this will always cause a transitive dependency to `dev-libs/icu`, in other cases this might not be true. It is good practice to always state exactly all _direct_ dependencies for a package so one can be sure that the dependencies will be pulled in.

Btw, this reasoning is similar to why paradigms like [IWYU](https://github.com/include-what-you-use/include-what-you-use?tab=readme-ov-file#instructions-for-users) exist in software development. Just the extra mental work of having to think about the implicit dependencies that are pulled in by the explicitly stated dependencies makes the ebuild harder to understand and can cause issues later on when the dependency implicitly pulling in the other dependency suddenly decides not to pull this dependency in anymore. So while this might make the ebuild shorter, I wouldn't consider it simpler and definitely not easier to maintain :sweat_smile:

Reply:
@NexAdn explained it perfectly. We have this also written in:
https://devmanual.gentoo.org/general-concepts/dependencies/#indirect-dependencies
(page bottom)